bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java

```diff
@@ -0,0 +1,687 @@
+            if (hasConnectivityIssue) {
+                if (thing.getStatusInfo().getStatusDetail() != ThingStatusDetail.COMMUNICATION_ERROR) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "@text/offline.clip2.communication-error.zigbee-connectivity-issue");
+                    // change all channel states, except the Zigbee channel itself, to undefined
+                    for (String channelId : supportedChannelIds) {
+                        if (!HueBindingConstants.CHANNEL_2_ZIGBEE_STATUS.equals(channelId)) {
+                            updateState(channelId, UnDefType.UNDEF);
+                        }
+                    }
+                }
+            } else if (thing.getStatus() != ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+                // one single refresh command will update all channels
+                Channel zigbeeChannel = thing.getChannel(HueBindingConstants.CHANNEL_2_ZIGBEE_STATUS);
+                if (Objects.nonNull(zigbeeChannel)) {
+                    handleCommand(zigbeeChannel.getUID(), RefreshType.REFRESH);
+                }
+            }
+        }
+    }
+
+    /**
+     * Execute a series of HTTP GET commands to fetch the resource data for all resources that contribute to the thing
+     * state.
+     *
+     * @throws ApiException if a communication error occurred.
+     * @throws AssetNotLoadedException if one of the assets is not loaded.
+     */
+    private void updateContributors() throws ApiException, AssetNotLoadedException {
+        logger.debug("updateContributors() called for {} contributors", contributorsCache.size());
+        ResourceReference reference = new ResourceReference();
+        for (Entry<String, Resource> entry : contributorsCache.entrySet()) {
+            updateResource(reference.setId(entry.getKey()).setType(entry.getValue().getType()));
+        }
+    }
+
+    /**
+     * Get all resources needed for building the thing state. Build the forward / reverse contributor lookup maps. Set
+     * up the final list of channels in the thing.
+     */
+    private void updateDependencies() {
+        if (!disposing && !updateDependenciesDone) {
+            logger.debug("updateDependencies() called");
+            try {
+                updateLookups();
+                updateContributors();
+                updateChannelList();
+                updateChannelItemLinksFromLegacy();
+                updateDependenciesDone = true;
+                if (!hasConnectivityIssue) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            } catch (ApiException e) {
+                logger.debug("updateDependencies() {}", e.getMessage(), e);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "@text/offline.communication-error");
```

Comment:
Use the exception message as detail message?

Reply:
Ok.